runtime/major_gc.c

```diff
@@ -635,6 +635,17 @@ static inline intnat diffmod (uintnat x1, uintnat x2)
   return (intnat) (x1 - x2);
 }
 
+/* Reset the work counter to be equal to the alloc counter. For use at
+ * times when we have disturbed the major GC from its usual pacing and
+ * tempo, e.g. at the end of a forced major collection.
+ */
+
+void caml_reset_major_pacing(void)
+{
+  work_counter = alloc_counter;
```

Comment:
This is enough to test with, but before we merge this it needs a more domain-aware implementation (it would be bad to _decrease_ alloc_counter here, which could happen if some other domain has run ahead of us)